From c2abd6ee5cdc848d8dfba33b6aa244289fc36a8a Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Tue, 22 Jul 2014 00:56:51 +0100
Subject: [PATCH 02/16] Undefine FORCEINLINE on MinGW-w64 in malloc.c.h

.. it contains extern and when combined with static leads to
'multiple storage classes in declaration specifiers'
---
 compat/nedmalloc/malloc.c.h | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/compat/nedmalloc/malloc.c.h b/compat/nedmalloc/malloc.c.h
index 6aeac26..22f3496 100644
--- a/compat/nedmalloc/malloc.c.h
+++ b/compat/nedmalloc/malloc.c.h
@@ -720,6 +720,13 @@ struct mallinfo {
   inlining are defined as macros, so these aren't used for them.
 */
 
+/* MinGW-w64 defines FORCEINLINE with extern, and when combined with
+   static, there's multiple storage classes in declaration specifiers.
+*/
+#if defined(__MINGW64_VERSION_MAJOR) && defined(FORCEINLINE)
+  #undef FORCEINLINE
+#endif
+
 #ifndef FORCEINLINE
   #if defined(__GNUC__)
 #define FORCEINLINE __inline __attribute__ ((always_inline))
-- 
2.1.2

